src/services/storage/drivers/postgres/index.ts

```diff
@@ -66,6 +76,29 @@ export class PostgresStorage implements StorageDriver {
 		])
 		await this.db.schema.dropSchemaIfExists(this.config.appName)
 	}
+
+	public async pushUpdates(): Promise<void> {
+		//run over the cache and dispatch db pushes
+		for (const [id, count] of Object.entries(this.countersCache.infoCount)) {
+			await this.db
+				.table<UrlInformation>('url_information')
+				.where('url_id', '=', id)
+				.increment('info_visit_count', count)
+				.update({ lastUsed: new Date().toISOString() })
+		}
+		for (const [id, count] of Object.entries(this.countersCache.visitCount)) {
+			await this.db
+				.table<UrlInformation>('url_information')
+				.where('url_id', '=', id)
+				.increment('url_visit_count', count)
+		}
+
+		//clear cache
+		this.countersCache = {
```

Comment:
You're clearing the cache after all requests are done, but the server keeps on running, taking requests and increasing the counters. The update should be a pure transaction.

Here's a pseudo-code describing the actions roughly needed:
```
const counters = this.countersCache
this.countersCache = {
	visitCount: {},
	infoCount: {},
}
try {
	await updateDbSql(Object.keys(counters).map(counter=>`UPDATE myTable SET....`).join(`;\n`))
} catch (err) {
	logger.error("Update counters failed!", err)
	// Revert countersCache changes since we failed the update
	this.countersCache = mergeCounters(this.countersCache, counters)
}
```